specs/agents/log-correlation.md

```diff
@@ -26,6 +26,10 @@ Log correlation is implemented by adding the following fields to a log document:
   - allows to correlate with an Error.
   - should be added to the MDC
 
+In addition, the `container.id` can be used as a fallback when `service.name` is not avaiable on the log documents.
+However, the APM agents are not expected to set it. It is expected to be set by filebeat when ingesting log
```

Comment:
This might be slightly confusing as agents do set the `container.id` in the [metadata](https://github.com/elastic/apm-server/blob/337cf8094e0ea06355fd55976c51e9567fa276fe/docs/spec/v2/metadata.json#L426) and all the metadata fields are applied to log records. That's also why agents don't actively need to set the service.name into the logs - it's already in the metadata.
However, the intake v2 metadata only applies to log sending, not to ecs reformatting. For the latter, agents do need to configure ECS logging libraries to add the service.name to logs. I agree that adding the `container.id` to logs seems unnecessary as the log collector is expected to do that.

Reply:
I think I will re-write it for better clarity, what do you think of something like this:

The `container.id` field can also be used as a fallback to provide service-level correlation, however agents are not expected to set it:
- log collector (filebeat) is expected to do that when ingesting logs.
- all data sent through agent intake implicitly provides `container.id` through metadata, which also includes the log events that may be sent to apm-server.

Reply:
done in https://github.com/elastic/apm/pull/766/commits/fbde920e5d467042ed41f3759c85519992e28923